host/class/uac/usb_host_uac/include/usb/uac_host.h

```diff
@@ -0,0 +1,374 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <wchar.h>
+#include <stdint.h>
+#include "esp_err.h"
+#include "uac.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief USB UAC HOST string descriptor maximal length
+ *
+ * The maximum possible number of characters in an embedded string is device specific.
+ * For USB devices, the maximum string length is 126 wide characters (not including the terminating NULL character).
+ * This is a length, which is available to upper level application during getting information
+ * of UAC Device with 'uac_host_get_device_info' call.
+ *
+ * To decrease memory usage 32 wide characters (64 bytes per every string) is used.
+*/
+#define UAC_STR_DESC_MAX_LENGTH              (32)
+#define FLAG_STREAM_SUSPEND_AFTER_START      (1 << 0)      /*!< suspend stream after start, then call uac_host_device_resume to resume */
+
+typedef struct uac_interface *uac_host_device_handle_t;    /*!< Logic Device Handle. Handle to a particular UAC interface */
+
+// ------------------------ USB UAC Host events --------------------------------
+/**
+ * @brief USB UAC HOST Driver event id
+*/
+typedef enum {
+    UAC_HOST_DRIVER_EVENT_RX_CONNECTED = 0x00,           /*!< UAC RX Device has been found in connected USB device */
+    UAC_HOST_DRIVER_EVENT_TX_CONNECTED,                  /*!< UAC TX Device has been found in connected USB device */
+} uac_host_driver_event_t;
+
+/**
+ * @brief USB UAC Device (Interface) event id
+*/
+typedef enum {
+    UAC_HOST_DEVICE_EVENT_RX_DONE = 0x00,                /*!< Data in the rx buffer is higher than the threshold */
+    UAC_HOST_DEVICE_EVENT_TX_DONE,                       /*!< Data in the tx buffer lower than the threshold */
+    UAC_HOST_DEVICE_EVENT_TRANSFER_ERROR,                /*!< UAC Device transfer error */
+    UAC_HOST_DRIVER_EVENT_DISCONNECTED,                  /*!< UAC Device has been disconnected */
+} uac_host_device_event_t;
+
+// ------------------------ USB UAC Host events callbacks -----------------------------
+/**
+ * @brief USB UAC driver event callback.
+ *
+ * @param[in] addr        USB Address of connected UAC device
+ * @param[in] iface_num   UAC Interface Number
+ * @param[in] event       UAC driver event
+ * @param[in] arg         User argument from UAC driver configuration structure
+*/
+typedef void (*uac_host_driver_event_cb_t)(uint8_t addr, uint8_t iface_num,
+        const uac_host_driver_event_t event, void *arg);
```

Comment:
For future extension of this API, it might be better to prepare a field for event specific data, eg.
``` c
struct {
    uac_host_device_event_t event_type;
    union {
        // For future extensions
    } event_data;
}
```

Reply:
will consider it after the second round of review